lamindb/models/artifact.py

```diff
@@ -261,9 +261,9 @@ def process_data(
     if key_suffix is not None and key_suffix != suffix and not is_replace:
         # consciously omitting a trailing period
         if isinstance(data, (str, Path, UPath)):  # UPathStr, spelled out
-            message = f"The suffix '{suffix}' of the provided path is inconsistent, it should be '{key_suffix}'"
+            message = f"Mismatch between data path suffix '{suffix}' and key suffix '{key_suffix}'."
```

Comment:
I think this is less easy to understand than the previous message both because it doesn't tell the user how to resolve and because it uses a non-established concept "data path".

How about the following:
```
f"The suffix '{suffix}' of the passed path is inconsistent with what it should be according to the target key: '{key_suffix}'"
```

Reply:
That's a bit verbose. I am also always in favor of suggesting a solution but here I thought the resolution would be obvious enough.

How about

`f"The suffix '{suffix}' must match target key suffix '{key_suffix}'"`

?

Reply:
Yes your suggestion is better up to that it should be `match the target key` instead of `match target key`, I think. For consistency one could use `passed key` rather than `target key`, because some people might be confused with what "target" means. It's not consistently referred to this way.